src/test/java/io/debezium/connector/yugabytedb/consistent/MessageTest.java

```diff
@@ -34,4 +61,157 @@ void comparatorTest() {
         assertEquals(1, commit.compareTo(insert));
     }
 
+    @Test
+    public void checkForBeginMessages() {
+        // Building a dummy row message
+        CdcService.RowMessage.Builder rowMessageBuilder = CdcService.RowMessage.newBuilder()
+                .setOp(CdcService.RowMessage.Op.BEGIN)
+                .setCommitTime(10L)
+                .setTransactionId(ByteString.EMPTY)
+                .setRecordTime(5L);
+        CdcService.RowMessage rowMessage = rowMessageBuilder.build();
+        CdcService.CDCSDKProtoRecordPB record = CdcService.CDCSDKProtoRecordPB.newBuilder()
+                .setRowMessage(rowMessage).build();
+
+        Message m = new Message.Builder().setRecord(record).setTabletId("dummyTabletId")
+                .setSnapshotTime(0).build();
+
+        assertTrue(Message.isBegin(m));
+    }
+
+    @Test
+    public void checkForCommitMessages() {
+        // Building a dummy row message
+        CdcService.RowMessage.Builder rowMessageBuilder = CdcService.RowMessage.newBuilder()
+                .setOp(CdcService.RowMessage.Op.COMMIT)
+                .setCommitTime(10L)
+                .setTransactionId(ByteString.EMPTY)
+                .setRecordTime(5L);
+        CdcService.RowMessage rowMessage = rowMessageBuilder.build();
+        CdcService.CDCSDKProtoRecordPB record = CdcService.CDCSDKProtoRecordPB.newBuilder()
+                .setRowMessage(rowMessage).build();
+
+        Message m = new Message.Builder().setRecord(record).setTabletId("dummyTabletId")
+                .setSnapshotTime(0).build();
+
+        assertTrue(Message.isCommit(m));
+    }
+
+    @Test
+    public void checkNeitherBeginNorCommit() {
+        // Building a dummy row message
+        CdcService.RowMessage.Builder rowMessageBuilder = CdcService.RowMessage.newBuilder()
+                .setOp(CdcService.RowMessage.Op.INSERT)
+                .setCommitTime(10L)
+                .setTransactionId(ByteString.EMPTY)
+                .setRecordTime(5L);
+        CdcService.RowMessage rowMessage = rowMessageBuilder.build();
+        CdcService.CDCSDKProtoRecordPB record = CdcService.CDCSDKProtoRecordPB.newBuilder()
+                .setRowMessage(rowMessage).build();
+
+        Message m1 = new Message.Builder().setRecord(record).setTabletId("dummyTabletId")
+                .setSnapshotTime(0).build();
+        Message m2 = new Message.Builder().setRecord(record).setTabletId("anotherDummyTablet")
+                .setSnapshotTime(0).build();
+
+        assertTrue(Message.notBeginCommit(m1, m2));
+    }
+
+    @DisplayName("Test for different combinations of compareTo")
+    @ParameterizedTest
```

Comment:
A minor usability request. `@ParameterizedTest` can take a display name as an argument. 

```
@ParameterizedTest(name = "{index}. Checking {0}")
...

private static Stream<Arguments> parameterSource() {
        return Stream.of(Arguments.of("Begin, Normal, Normal < Begin", Op.BEGIN, Op.INSERT, lowCommitTime, highCommitTime, 0, lowRecordTime, -1),
        ...
```

Will print something like "1. Begin, Normal, Normal < Begin" in test output and if something fails.